sw/device/examples/hello_world/BUILD

```diff
@@ -0,0 +1,47 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan.bzl", "OPENTITAN_CPU", "opentitan_binary")
+
+opentitan_binary(
+    name = "hello_world",
+    srcs = [
+        "hello_world.c",
+    ],
+    copts = [
+        "-nostdlib",
+        "-ffreestanding",
```

Comment:
It may be worth implying these in `ot_binary` by default.

Reply:
I like this idea, I assume we don't want/trust/have space to fit the things these options exclude so they're generally applicable (because it's probably less elegant to add it for all and turn it off for a few exceptions.)

Reply:
If we had wanted these for something it probably would show up in a hello-world example, or would be added in by an already large effort that involves work on the build system anyway.

Reply:
I have some important questions before I can get a PR that does that to the state where I'm comfortable enough to submit it for review. Given that's a nice separable change can we say it's outside the scope of this?

The questions are:
Is it ok to set these as a default so they only need to be written out if copts is otherwise set? Or should these be appended to the copts used in a rule such that new copts can be written without understanding the default behavior and these flags, and if so do we need a way to override and drop these flags or is it ok to always have them in opentitan_binaries?

If someone knew the answer to the first question was yes, I have a commit I could push to resolve this.

Reply:
Here's the draft PR for that. https://github.com/lowRISC/opentitan/pull/9142

Reply:
They should be appended. Not having these flags simply doesn't make sense for the OT target. The same goes for the linker options below.

Reply:
Agreed: lets move the `copts` and `linkopts` into the `opentitan_binary` rule.

Reply:
That should be done now